spec/app/domain/authentication/authn-oidc/authn_oidc_spec.rb

```diff
@@ -85,6 +85,51 @@ def request_body(request)
       end
 
       context "with valid id token" do
+        context "when webservice validation succeeds" do
+          subject do
+            input_ = Authentication::AuthenticatorInput.new(
+              authenticator_name: 'authn-oidc',
+              service_id:         'my-service',
+              account:            'my-acct',
+              username:           nil,
+              credentials:        request_body(authenticate_id_token_request),
+              origin:             '127.0.0.1',
+              request:            authenticate_id_token_request
+            )
+
+            ::Authentication::AuthnOidc::Authenticate.new(
+              enabled_authenticators:              authenticator_name,
+              token_factory:                       mocked_token_factory,
+              validate_user_can_access_webservice: mock_validate_user_can_access_webservice(validation_succeeded: true),
+              validate_webservice_is_whitelisted:  mock_validate_webservice_is_whitelisted(validation_succeeded: true),
+              validate_account_exists:             mocked_account_validator,
+              validate_origin:                     mocked_origin_validator,
+              verify_and_decode_token:             mocked_decode_and_verify_id_token,
+              audit_log:                           mocked_audit_logger
+            ).call(
+              authenticator_input: input_
+            )
+          end
+
+          it "returns a new access token" do
+            expect(subject).to equal(a_new_token)
+          end
+
+          it_behaves_like "raises an error when origin validation fails"
```

Comment:
Seems like the `it_behaves_like` list below are separate tests? Is `it_behaves_like` just `context` but nested?

Reply:
yes, they are separate tests. you can see their definition in the spec helper